### PR TITLE
Tour url

### PR DESCRIPTION
**Pull request by elguapogordo**
**Fixes issue #:** 

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling- no new features

**Description:**
add poro that accepts a collection of coordinates and returns a google maps url with walking directions between coordinates

**What does this solve?:**
tours endpoint now can return map with directions

**Notes:**
coordinate elements work with both strings and floats

- [x] All tests are passing

### CODE REVIEW

File: app/poro/tour_url.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+class TourUrl
+  attr_reader :coordinates, :origin, :center
```
Comment (elguapogordo) [inline app/poro/tour_url.rb:2]: attr_reader for testing purposes only, otherwise not needed

**Review by Gallup93 — APPROVED** [review]

File: app/poro/tour_url.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    end
+    avg_coord(center)
+  end
+
+  def avg_coord(center)
+    floats = center.map do |_k, v|
+      (v.min.to_f + v.max.to_f) / 2
+    end
+    '@' + floats.join(',') + ','
+  end
```
Comment (elguapogordo) [inline app/poro/tour_url.rb:33]: takes `[[lat,lon],[lat,lon],[lat,lon]]`
turns it to `{lat:[lat,lat,lat],lon:[lon,lon,lon]}
averages lat and lon min and max to center the map over the route
line 32 formats it for the url

File: app/poro/tour_url.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    @coordinates.map do |set|
+      "'" + set.join(',') + "'/"
+    end.join
+  end
+
+  def destinations
+    @coordinates.map do |set|
+      '!1m3!2m2!1d' + set.reverse.join('!2d')
+    end.join
+  end
```
Comment (elguapogordo) [inline app/poro/tour_url.rb:45]: for some reason lat/lon are reversed in this part of the url

File: app/poro/tour_url.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      (v.min.to_f + v.max.to_f) / 2
+    end
+    '@' + floats.join(',') + ','
+  end
+
+  def formatted_coords
+    @coordinates.map do |set|
+      "'" + set.join(',') + "'/"
+    end.join
+  end
```
Comment (elguapogordo) [inline app/poro/tour_url.rb:39]: all destinations pulled and formatted with this method

File: spec/poros/tour_url_spec.rb [hunk 5]
```diff
@@ -0,0 +1,13 @@
+require 'rails_helper'
+
+describe 'TourUrl', type: :poro do
+  it 'generates url string from coordinates' do
+    coordinates = [["39.757300", "-104.986918"], ["39.758697", "-104.985510"], ["39.759241", "-104.985322"], ["39.760942", "-104.983589"]]
+
+    url = TourUrl.new(coordinates).create_url
```
Comment (elguapogordo) [inline spec/poros/tour_url_spec.rb:7]: create_url would be private normally. but rspec can't test it that way

**Comment by stellakunzang** [comment]
Thanks @elguapogordo, these comments are very helpful!

**Pull request merged**